README.md

```diff
@@ -1,58 +1,164 @@
+const user = new User();
+user.firstName = "Homer";
+user.lastName = "Simpson";
+user.active = true;
+user.locations = ["Springfield"];
+user.projects = {
+  SPP: "Springfield Power Plant",
+};
+
+const mgr = new User();
+mgr.firstName = "Montgomery";
+mgr.lastName = "Burns";
+
+user.manager = mgr;
+```
+
+or by passing an initializer object to constructors (note that all fields are optional).
 
 ```typescript
-let pete = new Person({
-  name: "pete",
-  id: 123
+const user = new User({
+  firstName: "Homer",
+  lastName: "Simpson",
+  active: true,
+  locations: ["Springfield"],
+  projects: { SPP: "Springfield Power Plant" },
+  manager: {
+    // you can simply pass an initializer object for this nested message field
+    firstName: "Montgomery",
+    lastName: "Burns",
+  },
 });
+```
+
+The generated code also contains functions for serializing and deserializing your data using various formats.  You can serialize your data to and from bytes:
 
-let bytes = pete.toBinary();
-pete = Person.fromBinary(bytes);
-pete = Person.fromJsonString('{"name": "pete", "id": 123}');
+```typescript
+const bytes = user.toBinary()
+// ...
+const deserialized = User.fromBinary(bytes);    // deserialized is equal to the original user
 ```
 
-To learn more, have a look at a complete [code example](https://github.com/bufbuild/protobuf-es/tree/main/packages/protobuf-example), 
-the documentation for the [generated code](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md), 
-and the documentation for the [runtime API](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md).
+You can also serialize and deserialize with JSON:
 
+```typescript
+const json = user.toJson();
+// ...
+const deserialized = User.fromJson(json);    // deserialized is equal to the original user
+```
 
-### TypeScript
+Each generated message can also be stringified using `JSON.stringify` and recreated with that JSON string:
 
-The generated code is compatible with TypeScript **v4.1.2** or later, with the default compiler settings.
+```typescript
+const str = JSON.stringify(user);
+// ...
+const deserialized = User.fromJsonString(str);    // deserialized is equal to the original user
+```
 
+## Packages
 
-### Packages
+There are three main packages comprising the **Protobuf-ES** library:
 
-- [@bufbuild/protoc-gen-es](https://www.npmjs.com/package/@bufbuild/protoc-gen-es):
-  Provides the code generator plugin `protoc-gen-es` ([source](packages/protoc-gen-es)).   
-  The code it generates depends on `@bufbuild/protobuf`.
-- [@bufbuild/protobuf](https://www.npmjs.com/package/@bufbuild/protobuf):
-  The runtime library for the code generator plugin `protoc-gen-es` ([source](packages/protobuf)).
-- [@bufbuild/protoplugin](https://www.npmjs.com/package/@bufbuild/protoplugin):
-  Helps to create your own code generator plugin ([source](packages/protoplugin)).
+[**@bufbuild/protobuf**](https://www.npmjs.com/package/@bufbuild/protobuf)  ([Docs](packages/protobuf))
```

Comment:
We're linking to the very same README file twice here. Once on npmjs.com, once on GitHub. As a user, I appreciate a direct link to the _source_ for a package.

For the formatting, I think it makes sense to have a very similar looking "Ecosystem" section for [connect-go](https://github.com/bufbuild/connect-go#ecosystem) and [connect-web](https://github.com/bufbuild/connect-web#ecosystem). For the connect-web readme, we probably want to retain the bullet list for both "Ecosystem" and "Packages" for consistency. So for consistency for "Packages" between connect-web and protobuf-es, I think we should also stick to the bullet list?